arangod/Network/NetworkFeature.cpp

```diff
@@ -127,6 +153,8 @@ void NetworkFeature::validateOptions(std::shared_ptr<options::ProgramOptions> op
   if (_idleTtlMilli < 10000) {
     _idleTtlMilli = 10000;
   }
+
+  _maxInFlight = std::clamp(_maxInFlight, ::MinAllowedInFlight, ::MaxAllowedInFlight);
```

Comment:
Sometimes we do, sometimes we don't (for instance, see two lines above). I don't have strong feelings either way, but I think in case someone is setting it outside the allowed values here, they really have no idea what the value means. It should already be a hidden parameter, so they should only be setting it to something non-default in the case where we suggest it.

Reply:
Agreed, however on the long-run we may change the upper/lower bounds of this, and someone who set this parameter may not have seen this, so does not get it set as expected.
SO even for hidden parameters I am in favor of at least logging illegal input.